docs/language/src/builtins/builtin_elements.md

```diff
@@ -442,340 +446,309 @@ location specified by the `x` and `y` properties, using the control points speci
--   **`padding`** (_length_): the padding within the layout.
--   **`padding-left`**, **`padding-right`**, **`padding-top`** and **`padding-bottom`** (_length_):
-    override the padding in specific sides.
--   **`alignment`** (_FIXME enum_): Can be one of `stretch`, `center`, `start`, `end`,
-    `space-between`, `space-around`. Defaults to `stretch`. Matches the CSS flex.
+    // Rectangle with a border
+    Rectangle {
+        x: 70px;
+        y: 10px;
+        width: 50px;
+        height: 50px;
+        background: green;
+        border-width: 2px;
+        border-color: red;
+    }
 
-### Example
+    // Transparent Rectangle with a border and a radius
+    Rectangle {
+        x: 140px;
+        y: 10px;
+        width: 50px;
+        height: 50px;
+        border-width: 4px;
+        border-color: black;
+        border-radius: 10px;
+    }
 
-```slint
-export component Foo inherits Window {
-    width: 200px;
-    height: 100px;
-    HorizontalLayout {
-        spacing: 5px;
-        Rectangle { background: red; width: 10px; }
-        Rectangle { background: blue; min-width: 10px; }
-        Rectangle { background: yellow; horizontal-stretch: 1; }
-        Rectangle { background: green; horizontal-stretch: 2; }
+    // A radius of width/2 makes it a circle
+    Rectangle {
+        x: 210px;
+        y: 10px;
+        width: 50px;
+        height: 50px;
+        background: yellow;
+        border-width: 2px;
+        border-color: blue;
+        border-radius: self.width/2;
     }
 }
 ```
 
-## `GridLayout`
+## `TextInput`
 
-`GridLayout` places the elements in a grid. `GridLayout` adds properties to each item: `col`, `row`, `colspan`, `rowspan`.
-You can control the position of elements with `col` and `row`.
-If `col` or `row` is not specified, they are automatically computed such that the item is next to the previous item, in the same row.
-Alternatively, the item can be put in a `Row` element.
+The `TextInput` is a lower-level item that shows text and allows entering text.
+
+When not part of a layout, its width or height defaults to 100% of the parent element when not specified.
 
 ### Properties
 
--   **`spacing`** (_length_): The distance between the elements in the layout.
--   **`padding`** (_length_): the padding within the layout.
--   **`padding-left`**, **`padding-right`**, **`padding-top`** and **`padding-bottom`** (_length_):
-    override the padding in specific sides.
+-   **`color`** (_in_ _brush_): The color of the text (default value: depends on the style)
+-   **`font-family`** (_in_ _string_): The name of the font family selected for rendering the text.
+-   **`font-size`** (_in_ _length_): The font size of the text.
+-   **`font-weight`** (_in_ _int_): The weight of the font. The values range from 100 (lightest) to 900 (thickest). 400 is the normal weight.
+-   **`has-focus`** (_out_ _bool_): Set to `true` when this `TextInput` is focused and will receive [`KeyEvent`](builtin_structs.md#keyevent)s.
```

Comment:
I think with out properties we should explicitly name who sets it. I think that's really important. It helps to emphasise the nature of out properties. If the sentence starts with "Set to `true`" it could be misread as "When you set it to `true`...").

```suggestion
-   **`has-focus`** (_out_ _bool_): `TextInput` sets this to `true` when it is focused. Only then it receives [`KeyEvent`](builtin_structs.md#keyevent)s.
```